src/taipy/core/cycle/_cycle_manager.py

```diff
@@ -41,6 +41,20 @@ def _create(
         cls._set(cycle)
         return cycle
 
+    @classmethod
+    def _get_all(cls, version_number: Optional[str] = "all") -> List[Cycle]:
+        """
+        Returns all entities.
+        """
+        return cls._repository._load_all(version_number)
+
+    @classmethod
+    def _get_all_by(cls, by, version_number: Optional[str] = "all") -> List[Cycle]:
+        """
+        Returns all entities based on a criteria.
+        """
+        return cls._repository._load_all_by(by, version_number)
+
```

Comment:
This looks strange to me since the cycles are not versioned. Can't we just use the default implementation from the parent class ?

Reply:
On `get_all()` and `get_all_by()` on parent `_Manager` class, `"version_number"` would be `"current"` since most of the manager will need to get entities on the current version by default.

The only 2 exceptions are Cycle and _Version which doesn't have a version number to it, so the `"version_number"` would be `"all"`.
`"all"` basically means that we do not check for version when load from repository.

Reply:
OK.
Thx